src/mujincontrollerclient.cpp

```diff
@@ -132,23 +132,23 @@ void WebResource::Copy(const std::string& newname, int options, double timeout)
     throw MujinException("not implemented yet");
 }
 
-ObjectResource::ObjectResource(ControllerClientPtr controller, const std::string& pk) : WebResource(controller, "object", pk), pk(pk)
+ObjectResource::ObjectResource(ControllerClientPtr controller, const std::string& pk_) : WebResource(controller, "object", pk_), pk(pk_)
```

Comment:
Hmm, something is wrong with your linter, this is not shadowing. Not a big fan of appending `_` to variables.

Reply:
It does shadow member variable https://github.com/mujin/controllerclientcpp/blob/master/include/mujincontrollerclient/mujincontrollerclient.h#L819

We testing team considered prefixing member variable with "m_" but it costs too much.

FYI https://github.com/rdiankov/openrave/pull/1071 also contains such changes.

Reply:
`ObjectResource` class already has the member variable called `pk`, creating a member function parameter with exactly that name is shadowing according to g++ compiler we use. If you don't like underscores, I'm open to any other simple and transparent scheme for renaming shadowing variables that you like.

Reply:
It would have been nice to have the member prefixed with `_`, like `_pk`, so you can just use `_pk(pk)`, but since that's too much change (btw, why is it too much change?) ...

Reply:
We also don't use `m_` in our codebase.

Reply:
`member prefixed with _`, not arguments?

Reply:
initializer parameters are used usually once, in a constructor, very localized and easy to fix. member variables are used throughout the code, could be accessed in the dependent packages or who knows where -- hard to find all possible occurrences potentially scattered across multiple files. We have already had this discussion here: https://git.mujin.co.jp/test/test-townsquare/-/issues/1508